public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -0,0 +1,660 @@
+  group_wait?: string;
+  group_interval?: string;
+  repeat_interval?: string;
+};
+
+type InhertitableProperties = Pick<
+  Route,
+  'receiver' | 'group_by' | 'group_wait' | 'group_interval' | 'repeat_interval' | 'mute_time_intervals'
+>;
+
+interface PolicyComponentProps {
+  receivers?: Receiver[];
+  alertGroups?: AlertmanagerGroup[];
+  contactPointsState?: ReceiversState;
+  readOnly?: boolean;
+  inheritedProperties?: InhertitableProperties;
+  routesMatchingFilters?: RouteWithID[];
+
+  routeTree: RouteWithID;
+  currentRoute: RouteWithID;
+  alertManagerSourceName: string;
+  onEditPolicy: (route: RouteWithID, isDefault?: boolean) => void;
+  onAddPolicy: (route: RouteWithID) => void;
+  onDeletePolicy: (route: RouteWithID) => void;
+  onShowAlertInstances: (alertGroups: AlertmanagerGroup[], matchers?: ObjectMatcher[]) => void;
+}
+
+const Policy: FC<PolicyComponentProps> = ({
+  receivers = [],
+  contactPointsState,
+  readOnly = false,
+  alertGroups = [],
+  alertManagerSourceName,
+  currentRoute,
+  routeTree,
+  inheritedProperties,
+  routesMatchingFilters = [],
+  onEditPolicy,
+  onAddPolicy,
+  onDeletePolicy,
+  onShowAlertInstances,
+}) => {
+  const styles = useStyles2(getStyles);
+  const isDefaultPolicy = currentRoute === routeTree;
+
+  const permissions = getNotificationsPermissions(alertManagerSourceName);
+  const canEditRoutes = contextSrv.hasPermission(permissions.update);
+  const canDeleteRoutes = contextSrv.hasPermission(permissions.delete);
+
+  const contactPoint = currentRoute.receiver;
+  const continueMatching = currentRoute.continue ?? false;
+  const groupBy = currentRoute.group_by ?? [];
+  const muteTimings = currentRoute.mute_time_intervals ?? [];
+  const timingOptions: TimingOptions = {
+    group_wait: currentRoute.group_wait,
+    group_interval: currentRoute.group_interval,
+    repeat_interval: currentRoute.repeat_interval,
+  };
+
+  const matchers = normalizeMatchers(currentRoute);
+  const hasMatchers = Boolean(matchers && matchers.length);
+  const hasMuteTimings = Boolean(muteTimings.length);
+  const hasFocus = routesMatchingFilters.some((route) => route.id === currentRoute.id);
+
+  // gather warnings and errors here
+  const warnings: ReactNode[] = [];
+  const errors: ReactNode[] = [];
+
+  // if the route has no matchers, is not the default policy (that one has none) and it does not continue
+  // then we should warn the user that it's a suspicious setup
+  if (!hasMatchers && !isDefaultPolicy && !continueMatching) {
+    warnings.push(wildcardRouteWarning);
+  }
+
+  // if the receiver / contact point has any errors show it on the policy
+  const actualContactPoint = contactPoint ?? inheritedProperties?.receiver ?? '';
+  const contactPointErrors = contactPointsState ? getContactPointErrors(actualContactPoint, contactPointsState) : [];
+
+  contactPointErrors.forEach((error) => {
+    errors.push(error);
+  });
+
+  const childPolicies = currentRoute.routes ?? [];
```

Comment:
How about extracting some part of the logic from the component into a function when we can test it in isolation?